app/src/main/java/org/dslul/openboard/inputmethod/latin/settings/UserDictionaryAddWordContents.java

```diff
@@ -0,0 +1,363 @@
+    private final String mOldShortcut;
+    private final String mOldWeight;
+    private String mSavedWord;
+    private String mSavedShortcut;
+    private String mSavedWeight;
+
+    UserDictionaryAddWordContents(final View view, final Bundle args) {
+        mModeTitle = view.findViewById(R.id.user_dictionary_mode_title);
+        mWordEditText = view.findViewById(R.id.user_dictionary_add_word_text);
+        mWordEditText.requestFocus();
+        mShortcutEditText = view.findViewById(R.id.user_dictionary_add_shortcut);
+        mWeightEditText = view.findViewById(R.id.user_dictionary_add_weight);
+        mWeightEditText.setKeyListener(DigitsKeyListener.getInstance("0123456789"));
+        final Button deleteWordButton = view.findViewById(R.id.user_dictionary_delete_button);
+
+        final String word = args.getString(EXTRA_WORD);
+        if (null != word) {
+            mWordEditText.setText(word);
+            // Use getText in case the edit text modified the text we set. This happens when
+            // it's too long to be edited.
+            mWordEditText.setSelection(mWordEditText.getText().length());
+        }
+
+        final String shortcut = args.getString(EXTRA_SHORTCUT);
+        if (null != shortcut && null != mShortcutEditText) {
+            mShortcutEditText.setText(shortcut);
+        }
+        mOldShortcut = args.getString(EXTRA_SHORTCUT);
+
+        final String weight = args.getString(EXTRA_WEIGHT);
+        if (null != weight) {
+            mWeightEditText.setText(weight);
+        }
+
+        mMode = args.getInt(EXTRA_MODE);
+        if (mMode == MODE_EDIT) {
+            mModeTitle.setText(R.string.user_dict_mode_edit);
+            deleteWordButton.setVisibility(View.VISIBLE);
+        } else if (mMode == MODE_INSERT) {
+            mModeTitle.setText(R.string.user_dict_mode_insert);
+            deleteWordButton.setVisibility(View.INVISIBLE);
+        }
+
+        mOldWord = args.getString(EXTRA_WORD);
+        mOldWeight = args.getString(EXTRA_WEIGHT);
+        updateLocale(args.getString(EXTRA_LOCALE));
+    }
+
+    UserDictionaryAddWordContents(final View view, final UserDictionaryAddWordContents oldInstanceToBeEdited) {
+        mModeTitle = view.findViewById(R.id.user_dictionary_mode_title);
+        mWordEditText = view.findViewById(R.id.user_dictionary_add_word_text);
+        mShortcutEditText = view.findViewById(R.id.user_dictionary_add_shortcut);
+        mWeightEditText = view.findViewById(R.id.user_dictionary_add_weight);
+
+        mOldWord = oldInstanceToBeEdited.mSavedWord;
+        mOldShortcut = oldInstanceToBeEdited.mSavedShortcut;
+        mOldWeight = oldInstanceToBeEdited.mSavedWeight;
+        updateLocale(mLocale);
+    }
+
+    // locale may be null, this means default locale
+    // It may also be the empty string, which means "all locales"
+    void updateLocale(final String locale) {
+        mLocale = null == locale ? Locale.getDefault().toString() : locale;
```

Comment:
`Locale.getDefault` "_Gets the current value of the default locale for this instance of the Java Virtual Machine._"
Are you sure this is the default you want here?

Reply:
I didn't change this part because now `mLocale` will never be null.
Would you like an update?

Reply:
> Would you like an update?

That depends on which default you mean with "this means default locale". When you want the "main system locale" as seen by the app (I think it can be changed per App in newer Android versions, not sure what exactly happens then), the reading from `Configuration` or `SubtypeSettings` might be preferable.

Reply:
Done with commit 26e352a